Makefile

```diff
@@ -32,24 +32,57 @@ DIALYZER_APPS = kernel stdlib sasl erts ssl tools os_mon runtime_tools crypto in
 	xmerl webtool snmp public_key mnesia eunit syntax_tools compiler
 DIALYZER_FLAGS = -Wno_return
 TEST_PLT = .yokozuna_test_dialyzer_plt
+RIAK_TEST_PATH = riak_test
 
 include tools.mk
 
 ${TEST_PLT}: compile-riak-test
-ifneq (,$(RIAK_TEST_PATH))
-ifneq (,$(wildcard $(TEST_PLT)))
-	dialyzer --check_plt --plt $(TEST_PLT) && \
-		dialyzer --add_to_plt --plt $(TEST_PLT) --apps edoc --output_plt $(TEST_PLT) ebin $(RIAK_TEST_PATH)/ebin $(RIAK_TEST_PATH)/deps/riakc/ebin ; test $$? -ne 1
-else
-	dialyzer --build_plt --apps edoc --output_plt $(TEST_PLT) ebin $(RIAK_TEST_PATH)/ebin $(RIAK_TEST_PATH)/deps/riakc ; test $$? -ne 1
-endif
-else
-	@echo "Set RIAK_TEST_PATH"
-	exit 1
-endif
-
-dialyzer_rt: deps ${PLT} ${LOCAL_PLT} $(TEST_PLT)
-	dialyzer -Wno_return --plts $(PLT) $(LOCAL_PLT) $(TEST_PLT) -c riak_test/ebin
+	@if [ -d $(RIAK_TEST_PATH) ]; then \
+		if [ -f $(TEST_PLT) ]; then \
+			dialyzer --check_plt --plt $(TEST_PLT) $(RIAK_TEST_PATH)/ebin && \
+			dialyzer --add_to_plt --plt $(TEST_PLT) --apps edoc --output_plt $(TEST_PLT) ebin $(RIAK_TEST_PATH)/ebin ; test $$? -ne 1; \
+		else \
+			dialyzer --build_plt --apps edoc --output_plt $(TEST_PLT) ebin $(RIAK_TEST_PATH)/ebin ; test $$? -ne 1; \
+		fi \
+	fi
+
+dialyzer-rt-run:
+	@echo "==> $(shell basename $(shell pwd)) (dialyzer_rt)"
+	@PLTS="$(PLT) $(LOCAL_PLT) $(TEST_PLT)"; \
+	if [ -f dialyzer.ignore-warnings ]; then \
+		if [ $$(grep -cvE '[^[:space:]]' dialyzer.ignore-warnings) -ne 0 ]; then \
+			echo "ERROR: dialyzer.ignore-warnings contains a blank/empty line, this will match all messages!"; \
+			exit 1; \
+		fi; \
+		dialyzer $(DIALYZER_FLAGS) --plts $${PLTS} -c $(RIAK_TEST_PATH)/ebin > dialyzer_warnings ; \
+		cat dialyzer.ignore-warnings \
+		| sed -E 's/^([^:]+:)[^:]+:/\1/' \
+		| sort \
+		| uniq -c \
+		| sed -E '/.*\.erl: /!s/^[[:space:]]*[0-9]+[[:space:]]*//' \
+		> dialyzer.ignore-warnings.tmp ; \
+		egrep -v "^[[:space:]]*(done|Checking|Proceeding|Compiling)" dialyzer_warnings \
+		| sed -E 's/^([^:]+:)[^:]+:/\1/' \
+		| sort \
+		| uniq -c \
+		| sed -E '/.*\.erl: /!s/^[[:space:]]*[0-9]+[[:space:]]*//' \
+		| grep -F -f dialyzer.ignore-warnings.tmp -v \
+		| sed -E 's/^[[:space:]]*[0-9]+[[:space:]]*//' \
+		| sed -E 's/([]\^:+?|()*.$${}\[])/\\\1/g' \
+		| sed -E 's/(\\\.erl\\\:)/\1[[:digit:]]+:/g' \
+		| sed -E 's/^(.*)$$/^[[:space:]]*\1$$/g' \
+		> dialyzer_unhandled_warnings ; \
+		rm dialyzer.ignore-warnings.tmp; \
+		if [ $$(cat dialyzer_unhandled_warnings | wc -l) -gt 0 ]; then \
+		    egrep -f dialyzer_unhandled_warnings dialyzer_warnings ; \
+			found_warnings=1; \
+	    fi; \
+		[ "$$found_warnings" != 1 ] ; \
+	else \
+		dialyzer -Wno_return $(DIALYZER_FLAGS) --plts $${PLTS} -c $(RIAK_TEST_PATH)/ebin; \
+	fi
+
+dialyzer_rt: deps ${PLT} ${LOCAL_PLT} $(TEST_PLT) dialyzer-rt-run
```

Comment:
we now have a dialyzer command to run for test specs (locally)... please be aware that some test modules that get rebased-in from 2.0 may fail when running `make dialyzer_rt` ^^ @jvoegele